docs/reference/scripting/using.asciidoc

```diff
@@ -1,198 +1,243 @@
+script valuation. There's a lot happening here, but we'll break it down the
+components to understand them individually. For now, you only need to
+understand that this script takes `my_field` and operates on it.
 
+[source,console]
+----
 GET my-index-000001/_search
 {
   "script_fields": {
     "my_doubled_field": {
-      "script": {
-        "lang":   "expression",
-        "source": "doc['my_field'] * multiplier",
+      "script": { <1>
+        "source": "doc['my_field'].value * params['multiplier']", <2>
         "params": {
           "multiplier": 2
         }
       }
     }
   }
 }
--------------------------------------
-
-[discrete]
-=== Script parameters
-
-`lang`::
-
-    Specifies the language the script is written in.  Defaults to `painless`.
-
-
-`source`, `id`::
+----
+// TEST[continued]
+<1> `script` object
+<2> `script` source
 
-    Specifies the source of the script.  An `inline` script is specified
-    `source` as in the example above. A `stored` script is specified `id`
-    and is retrieved from the cluster state (see <<modules-scripting-stored-scripts,Stored Scripts>>).
+The `script` is a standard JSON object that defines scripts under most APIs
+in {es}. This object requires `source` to define the script itself. The
+script doesn't specify a language, so it defaults to Painless.
 
+[discrete]
+[[prefer-params]]
+=== Use parameters in your script
 
-`params`::
+The first time {es} sees a new script, it compiles it and stores the
+compiled version in a cache. Compilation can be a heavy process. If you're
+passing variables into a script, pass them in as named `params` instead of
+hard-coding values into the script itself.
 
-    Specifies any named parameters that are passed into the script as
-    variables.
+For example, in the previous script, we could have just hard coded values and
+written a script that is seemingly less complex. We could just retrieve the
+first value for `my_field` and then multiply it by `2`:
 
-[IMPORTANT]
-[[prefer-params]]
-.Prefer parameters
-========================================
+[source,painless]
+----
+"source": "return doc['my_field'].value * 2"
+----
 
-The first time Elasticsearch sees a new script, it compiles it and stores the
-compiled version in a cache. Compilation can be a heavy process.
+Though it works, this solution is pretty inflexible. We have to modify the
+script source to change the multiplier, and {es} has to recompile the script
+every time that the multiplier changes.
 
-If you need to pass variables into the script, you should pass them in as
-named `params` instead of hard-coding values into the script itself.  For
-example, if you want to be able to multiply a field value by different
-multipliers, don't hard-code the multiplier into the script:
+Instead of hard-coding values, use named `params` to make scripts flexible, and
+also reduce compilation time when the script runs. Our script declares a
```

Comment:
> Our script declares a parameter named `multiplier` and assigns it a value that we can access it anywhere in our script. Also, the script is compiled only once.

Remove.

"Declared" is an odd word for parameters. Additionally they are not available in user functions, unless passed in to the function, and thus we cannot "access it anywhere in our script."

Add

> Now you can change the `multiplier` parameter without Elasticsearch recompiling the script.

Reply:
I'll change this to read:
>You can now make changes to the `multiplier` parameter without {es} recompiling the script.